### PR TITLE
Make CUDNN an alias of MIOPEN for HIP ops

### PR DESCRIPTION
**Pull request by bddppq**
This is mostly for reusing all the cudnn test cases in our python operator_tests.

### CODE REVIEW

**Comment by bddppq** [comment]
triggered tests https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/188/

**Comment by bddppq** [comment]
@petrex @ashishfarmer There are some tests failing for the MIOPEN conv ops, could you help taking a look? [error message](https://gist.github.com/bddppq/9d8db91f2d77ce58e4668cdea1b32739), [detailed jenkins log](https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-test/33/consoleFull)

**Comment by petrex** [comment]
@Farmer, Ashish<mailto:Ashish.Farmer@amd.com> has a PR for group conv? Maybe we need to land that one first?

**Comment by ashishfarmer** [comment]
This is a regression that was caused by my earlier check in for group conv optimizations. The changes in PR#12273 will fix this issue. Could we land that one first please

**Comment by petrex** [comment]
#12273 @bddppq

**Comment by bddppq** [comment]
@ashishfarmer @petrex ah I see. landing #12273

**Comment by bddppq** [comment]
New tests triggered https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/265/

**Comment by bddppq** [comment]
@petrex @ashishfarmer there is a new error in test_convolution_layout https://gist.github.com/bddppq/b40b12637acb9b2554412525d1e78094

**Comment by petrex** [comment]
I am guessing { name: "order" s: "NHWC" }
@Farmer, Ashish<mailto:Ashish.Farmer@amd.com> Can you pls take a look ? thx

**Comment by ashishfarmer** [comment]
@bddppq, It is trying to run NHWC using MIOpen. Currently we don't support NHWC layout in MIOpen.
In the conv_test.py, there needs a check
```
if _run_in_hip(gc, dc) and (engine == 'CUDNN') and (order != 'NCHW'):
            return
```
similar to the checks in spatial_bn_op_test.py

**Comment by bddppq** [comment]
@ashishfarmer Thanks for looking! Could you take this PR over to fix all the failed miopen tests?

**Comment by ashishfarmer** [comment]
Sure thing. Will check and enable appropriate tests for miopen

File: caffe2/python/operator_test/conv_test.py [hunk 1]
```diff
@@ -462,8 +468,12 @@ def test_convolution_layout(self, op_type, stride, pad, kernel, dilation,
 
         for order in ["NCHW", "NHWC"]:
             engine_list = ['']
-            if _cudnn_supports(dilation=(dilation > 1), nhwc=(order == 'NHWC')):
-                engine_list.append('CUDNN')
+            if _run_in_hip(gc, dc):
+                if order == 'NCHW':
+                    engine_list.append('CUDNN')
```
Comment (bddppq) [inline caffe2/python/operator_test/conv_test.py]: Since we have control here, we can append `MIOPEN`.

**Comment by bddppq** [comment]
Triggered tests https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/738/

**Comment by ashishfarmer** [comment]
@bddppq, triggered tests after the commits: https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/1015/
The test that is failing is RNN cell test, and may have nothing to do with CUDNN aliasing. We are looking into the failure, but as far as changes for this PR are concerned, it looks good.

**Comment by petrex** [comment]
failing test : operator_test.rnn_cell_test.RNNCellTest.**test_lstm_with_dot_attention**_different_dim (from pytest)

IIRC , miopen does not support attention in lstm at this moment. Shall we skip this test? 
@rohithkrn pls chime in.

**Comment by rohithkrn** [comment]
This test doesn't use miopen implementation of LSTM.

File: caffe2/python/operator_test/conv_test.py [hunk 2]
```diff
@@ -19,6 +19,9 @@
 import unittest
 import os
 
+def _run_in_hip(gc, dc):
```
Comment (bddppq) [inline caffe2/python/operator_test/conv_test.py]: as this appears multiple times in the codebase now, could you put it in a test_util file? (say hip_test_util.py)

File: caffe2/python/operator_test/spatial_bn_op_test.py [hunk 3]
```diff
@@ -79,11 +81,13 @@ def reference_spatialbn_test(X, scale, bias, mean, var):
...
            inplace=st.booleans(),
            engine=st.sampled_from(["", "CUDNN"]),
-           # Currently HIP SpatialBN only supports 2D
-           **hu.gcs_no_hip)
+           **hu.gcs)
     def test_spatialbn_test_mode_1d(
             self, size, input_channels, batch_size, seed, order, epsilon,
             inplace, engine, gc, dc):
+        # Currently MIOPEN SpatialBN only supports 2D
+        if _run_in_hip(gc, dc) and engine != "":
```
Comment (bddppq) [inline caffe2/python/operator_test/spatial_bn_op_test.py]: nit: better to compare it with `engine == "CUDNN"`

File: caffe2/python/operator_test/spatial_bn_op_test.py [hunk 4]
```diff
@@ -255,11 +259,13 @@ def test_spatialbn_train_mode_gradient_check(
...
            engine=st.sampled_from(["", "CUDNN"]),
-           # Currently HIP SpatialBN only supports 2D
-           **hu.gcs_no_hip)
+           **hu.gcs)
     def test_spatialbn_train_mode_gradient_check_1d(
             self, size, input_channels, batch_size, seed, order, epsilon,
             momentum, engine, gc, dc):
+        # Currently MIOPEN SpatialBN only supports 2D
+        if _run_in_hip(gc, dc) and engine != "":
+            return
```
Comment (bddppq) [inline caffe2/python/operator_test/spatial_bn_op_test.py]: use assume

File: caffe2/python/operator_test/spatial_bn_op_test.py [hunk 5]
```diff
@@ -30,11 +30,13 @@ class TestSpatialBN(serial.SerializedTestCase):
...
            engine=st.sampled_from(["", "CUDNN"]),
-           # Currently HIP SpatialBN only supports 2D
-           **hu.gcs_no_hip)
+           **hu.gcs)
     def test_spatialbn_test_mode_3d(
             self, size, input_channels, batch_size, seed, order, epsilon,
             inplace, engine, gc, dc):
+        # Currently MIOPEN SpatialBN only supports 2D
+        if _run_in_hip(gc, dc) and engine != "":
+            return
```
Comment (bddppq) [inline caffe2/python/operator_test/spatial_bn_op_test.py]: use assume

**Comment by bddppq** [comment]
triggered c2 rocm tests https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/3118/

**Pull request closed without merging**